sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -998,6 +998,22 @@ class Analyzer(
   }
 
   /**
+   * Replace unresolved expressions in grouping keys with resolved ones in SELECT clauses.
+   */
+  object ResolveAggAliasInGroupBy extends Rule[LogicalPlan] {
+
+    override def apply(plan: LogicalPlan): LogicalPlan = plan.resolveOperators {
+      case agg @ Aggregate(groups, aggs, child)
+          if conf.groupByAliases && child.resolved && aggs.forall(_.resolved) &&
```

Comment:
I got debug info and it seems `ResolveAggAliasInGroupBy` was firstly applied in that case?
```
scala> sql("SELECT *, count(1) FROM (select 1 AS a, 1 AS b) GROUP BY a, b").show

=== Applying Rule org.apache.spark.sql.catalyst.analysis.Analyzer$ResolveReferences ===
!'Aggregate ['a, 'b], [*, unresolvedalias('count(1), None)]   'Aggregate ['a, 'b], [a#0, b#1, unresolvedalias('count(1), None)]
 +- Project [1 AS a#0, 1 AS b#1]                              +- Project [1 AS a#0, 1 AS b#1]
    +- OneRowRelation$                                           +- OneRowRelation$

<-- ResolveAggAliasInGroupBy applied -->

=== Applying Rule org.apache.spark.sql.catalyst.analysis.Analyzer$ResolveFunctions ===
!'Aggregate ['a, 'b], [a#0, b#1, unresolvedalias('count(1), None)]   'Aggregate ['a, 'b], [a#0, b#1, unresolvedalias(count(1), None)]
 +- Project [1 AS a#0, 1 AS b#1]                                     +- Project [1 AS a#0, 1 AS b#1]
    +- OneRowRelation$                                                  +- OneRowRelation$

=== Applying Rule org.apache.spark.sql.catalyst.analysis.Analyzer$ResolveAliases ===
!'Aggregate ['a, 'b], [a#0, b#1, unresolvedalias(count(1), None)]   'Aggregate ['a, 'b], [a#0, b#1, count(1) AS count(1)#3L]
 +- Project [1 AS a#0, 1 AS b#1]                                    +- Project [1 AS a#0, 1 AS b#1]
    +- OneRowRelation$                                                 +- OneRowRelation$

=== Applying Rule org.apache.spark.sql.catalyst.analysis.Analyzer$ResolveReferences ===
!'Aggregate ['a, 'b], [a#0, b#1, count(1) AS count(1)#3L]   Aggregate [a#0, b#1], [a#0, b#1, count(1) AS count(1)#3L]
 +- Project [1 AS a#0, 1 AS b#1]                            +- Project [1 AS a#0, 1 AS b#1]
    +- OneRowRelation$                                         +- OneRowRelation$

<-- ResolveAggAliasInGroupBy applied -->
```